### ISSUE TITLE
Getting all clients apps

### ISSUE THREAD

**Issue by xven0mxz** [jseerden/IntuneBackupAndRestore#38]
Hi Guys.

When i try to make a back-up of the clients apps with Invoke-IntuneBackupClientApp.ps1 i will get all application that are not installed and visible in my MEM environment. I have this issues in 5 tenants. So i think it's something with the script of graph.

These are the standard apps that I'm retrieving. with : $clientApps = Get-DeviceAppManagement_MobileApps | Get-MSGraphAllPages from the Invoke-IntuneBackupClientApp.ps1 script.

Adobe Acrobat Reader
Adobe Acrobat Reader
Adobe Acrobat Reader for Intune (Deprecated)
Adobe Acrobat Reader for Intune (Deprecated)
Azure Information Protection
Azure Information Protection
BlueJeans Video Conferencing
BlueJeans Video Conferencing
Board Papers
Box
Box — Cloud Content Management
Box for EMM
Breezy for Intune
CellTrust SL2™ for Intune
CellTrust SL2™ for Intune
Cisco Jabber for Intune
Cisco Jabber for Intune
Citrix ShareFile for Intune
Citrix ShareFile for Intune
Comfy
Comfy
Cortana
Cortana
Dynamics 365 Remote Assist
Dynamics 365 Remote Assist
FactSet
FactSet
Field Service (Dynamics 365)
Field Service (Dynamics 365)
Field Service Mobile
Field Service Mobile
FleetSafer
FleetSafer
Fuze Mobile for Intune
Fuze Mobile for Intune
Hearsay Relate for Intune
Hearsay Relate for Intune
iBabs For Intune
ISEC7 Mobile Exchange Delegate for Intune
Lexmark Mobile Print Intune
Meetio
Meetio
MentorcliQ
MentorcliQ
M-Files for Intune
M-Files for Intune
Microsoft 365 Admin
Microsoft 365 Admin
Microsoft Bookings
Microsoft Bookings
Microsoft Connections
Microsoft Connections
Microsoft Dynamics 365
Microsoft Dynamics 365 for phones
Microsoft Dynamics 365 for phones
Microsoft Dynamics 365 for tablets
Microsoft Edge
Microsoft Edge
Microsoft Excel
Microsoft Excel
Microsoft Invoicing
Microsoft Invoicing
Microsoft Kaizala
Microsoft Kaizala
Microsoft Launcher
Microsoft Lists
Microsoft OneDrive
Microsoft OneDrive
Microsoft OneNote
Microsoft OneNote
Microsoft Outlook
Microsoft Outlook
Microsoft Planner
Microsoft Planner
Microsoft Power Apps
Microsoft Power BI
Microsoft Power BI
Microsoft PowerPoint
Microsoft PowerPoint
Microsoft SharePoint
Microsoft SharePoint
Microsoft StaffHub
Microsoft StaffHub
Microsoft Stream
Microsoft Stream
Microsoft Teams
Microsoft Teams
Microsoft To-Do
Microsoft To-Do
Microsoft Visio Viewer
Microsoft Whiteboard
Microsoft Word
Microsoft Word
MultiLine for Intune
MultiLine for Intune
Nine Work for Intune
Notate for Intune
Notate for Intune
Now® Mobile - Intune
Now® Mobile - Intune
Omnipresence Go
Omnipresence Go
Outlook Groups
Outlook Groups
Power Apps
Power Automate
Power Automate
PrinterOn for Microsoft
PrinterOn for Microsoft
Qlik Sense Mobile
Qlik Sense Mobile
Remote Desktop
Remote Desktop
Senses
ServiceNow® Agent - Intune
ServiceNow® Agent - Intune
ServiceNow® Onboarding -Intune
ServiceNow® Onboarding -Intune
Skype for Business
Skype for Business
Smartcrypt For Intune
Speaking Email
STid Mobile ID
STid Mobile ID
Tableau Mobile for Intune
Tableau Mobile for Intune
Vera for Intune
Work Folders
Work Folders
Yammer
Yammer
ZERØ - email for attorneys
Zero for Intune
Zoom for Intune
Zoom for Intune

But when i look in the MEM portal there are no application. How is this possible? 
![image](https://user-images.githubusercontent.com/66790322/118828748-d530b180-b8bd-11eb-8538-e04cc5278a94.png)

**Comment by petter-bomban** [comment]
This behaviour comes from the cmdlet "Get-DeviceAppManagement_MobileApps" from the "Microsoft.Graph.Intune" module.

**Comment by jonod8698** [comment]
@xven0mxz I've worked around this by only including assigned apps. If this could be included as a switch in the module it'd be great. I just started learning PowerShell so it's a bit out of my reach at the moment.

In this directory "C:\Program Files\WindowsPowerShell\Modules\IntuneBackupAndRestore\2.1.1\Public"

I added Where-Object {$_.isAssigned -eq 1} in Invoke-IntuneBackupClientApp.ps1 and Invoke-IntuneBackupClientAppAssignment.ps1

` Get-DeviceAppManagement_MobileApps | Where-Object {$_.isAssigned -eq 1} | Get-MSGraphAllPages `

Make sure you restart powershell after.
